File: auth-center/auth-center-infra/src/main/java/com/jmsoftware/maf/authcenter/user/payload/GetUserStatusPayload.java
package com.jmsoftware.maf.authcenter.user.payload;

import com.jmsoftware.maf.common.domain.authcenter.user.UserStatus;
import com.jmsoftware.maf.common.domain.authcenter.user.UserStatus2;
import com.jmsoftware.maf.springcloudstarter.validation.annotation.ValidEnumValue;
import lombok.Data;

/**
 * <h1>GetUserStatusPayload</h1>
 * <p>
 * Change description here.
 *
 * @author Johnny Miller (鍾俊), email: johnnysviva@outlook.com
 * @date 6/6/21 4:31 PM
 **/
@Data
public class GetUserStatusPayload {
    @ValidEnumValue(targetEnum = UserStatus.class)
    private Byte status;
    @ValidEnumValue(targetEnum = UserStatus2.class)
    private Byte status2;
}

File: auth-center/auth-center-infra/src/main/kotlin/com/jmsoftware/maf/authcenter/user/payload/GetUserStatusPayload.kt
package com.jmsoftware.maf.authcenter.user.payload

import com.jmsoftware.maf.common.domain.authcenter.user.UserStatus
import javax.validation.constraints.NotNull

/**
 * # GetUserStatusPayload
 *
 * Change description here.
 *
 * @author Johnny Miller (锺俊), e-mail: johnnysviva@outlook.com, date: 4/12/22 12:50 PM
 */
class GetUserStatusPayload {
    @NotNull
    lateinit var status: UserStatus
}
